### PR TITLE
Extendable text block field input 

### LINKED ISSUE #3016: Extending blocks type "text" adds p tags to the input
**Description**  
I’m currently extending the text block type:
```
text:
  extends: blocks/text
  fields:
    text:
      type: textarea
```
But when editing the block, there are p tags added to the content. Even while writing in the input field (while the right side modal of the block is opened) there are p tags added to each typed character.

**Expected behavior**  
p tags shouldn't be added to the input field.

**Kirby version**  
3.5

**Desktop**  
 - OS: macOS 10.15.7
 - Browser: Chrome
 - Version: 87

### PR DESCRIPTION
**Pull request by afbora**
## This PR …

Text block input can now be extended more smoothly. It supports all fields.

````yaml
text:
  extends: blocks/text
  fields:
    text:
      type: textarea
      size: large
      spellcheck: false
      buttons: false
````

#### Enhancements

- Now text block field input extendable #3016

### Breaking changes

None

## Ready?

- [ ] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] Tests and checks all pass

### For review team

- [x] Add changes to release notes draft in Notion
- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)

### CODE REVIEW

**Comment by afbora** [comment]
If the PR is appropriate, we can publish it in version `3.7.4`.

**Comment by afbora** [comment]
Maybe we should do as in this PR for all core blocks or find a different general solution. Another extending core block issue: https://github.com/getkirby/kirby/issues/4560

File: panel/src/components/Forms/Blocks/Types/Text.vue [hunk 1]
```diff
@@ -1,10 +1,8 @@
 <template>
-	<k-writer
+	<component
+		:is="component"
 		ref="input"
-		:inline="textField.inline"
-		:marks="textField.marks"
-		:nodes="textField.nodes"
-		:placeholder="textField.placeholder"
+		v-bind="textField || $props"
```
Comment (distantnative) [inline panel/src/components/Forms/Blocks/Types/Text.vue]: What's the fallback to `$props` for? Isn't the only prop `endpoints` and wouldn't fit?

Comment (afbora) [reply panel/src/components/Forms/Blocks/Types/Text.vue]: Yes. `textField` can be empty as `{}`. Thus fallback to default props.

Comment (distantnative) [reply panel/src/components/Forms/Blocks/Types/Text.vue]: But `$props` is just `{ endpoint : ...}` for `Blocks/Types/Text.vue`. Passing that to the component doesn't make sense, does it?

Comment (afbora) [reply panel/src/components/Forms/Blocks/Types/Text.vue]: Yes I know. But if it doesn't make sense to you, we can remove it. I was afraid of breaking things without `endpoints`.

Comment (distantnative) [reply panel/src/components/Forms/Blocks/Types/Text.vue]: But we weren't passing `endpoints` before this PR - do you think it would break?

Comment (afbora) [reply panel/src/components/Forms/Blocks/Types/Text.vue]: I found the source of `endpoints` implementation: https://github.com/getkirby/kirby/commit/81d489d19e5deeab406867bb47eb55cdcd1d112b . I think we doesn't need `endpoints` anymore.

File: panel/src/components/Forms/Blocks/Types/Text.vue [hunk 2]
```diff
@@ -17,10 +15,23 @@
...
 export default {
-	props: {
-		endpoints: Object
-	},
 	computed: {
+		component() {
+			const type = this.textField.type;
+			if (type === "writer") {
+				return "k-writer";
+			}
```
Comment (distantnative) [inline panel/src/components/Forms/Blocks/Types/Text.vue]: We could simplify this further and remove this if statement. Then the code would check for `k-writer-input` which does not exist and still fall back to `k-writer`.

Comment (afbora) [reply panel/src/components/Forms/Blocks/Types/Text.vue]: I thought about it but wanted to avoid any extra checking 👍

**Comment by distantnative** [comment]
@afbora what do you think, is there a general solution or should we adapt each block to be better extendable?

**Comment by afbora** [comment]
The general solution would be: show the default block view (no editable for preview) if it's not the expected type: https://github.com/getkirby/kirby/issues/3016#issuecomment-752271272 But I have no idea about its implementation.

If we do it separately for each block, we will have a richer extending. As in this PR.

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**